app/main.py

```diff
@@ -1 +1,65 @@
-# write your code here
+from __future__ import annotations
+
+
+class AliveList(list):
+    def __repr__(self) -> str:
+        animals = (
+            f"{{Name: {animal.name}, "
+            f"Health: {animal.health}, "
+            f"Hidden: {animal.hidden}}}"
+            for animal in self
+        )
+        return f"[{', '.join(animals)}]"
+
+
+class Animal:
+    alive: AliveList = AliveList()
+
+    def __init__(
+        self,
+        name: str,
+        health: int = 100,
+    ) -> None:
+        self.name: str = name
+        self.health: int = health
+        self.hidden: bool = False
+        # FIX: Only add to alive list if health > 0
+        if self.health > 0:
+            Animal.alive.append(self)
+
+    def die(
+        self,
+    ) -> None:
+        if self in Animal.alive:
+            Animal.alive.remove(self)
+
+    def __repr__(
+        self,
+    ) -> str:
```

Comment:
```suggestion
    def __repr__(self) -> str:
```
If you can write this in one line, it's better to do so.